airflow/contrib/operators/livy_operator.py

```diff
@@ -0,0 +1,175 @@
+    :param files: files to be used in this session.
+    :type files: list
+    :param driver_memory: Amount of memory to use for the driver process  string.
+    :type driver_memory: str
+    :param driver_cores: Number of cores to use for the driver process int.
+    :type driver_cores: str
+    :param executor_memory: Amount of memory to use per executor process  string.
+    :type executor_memory: str
+    :param executor_cores: Number of cores to use for each executor  int.
+    :type executor_cores: str
+    :param num_executors: Number of executors to launch for this session  int.
+    :type num_executors: str
+    :param archives: Archives to be used in this session.
+    :type archives: list
+    :param queue: The name of the YARN queue to which submitted string.
+    :type queue: str
+    :param name: The name of this session  string.
+    :type name: str
+    :param conf: Spark configuration properties.
+    :type conf: dict
+    :param proxy_user: User to impersonate when running the job.
+    :type proxy_user: str
+    :param livy_conn_id: reference to a pre-defined Livy Connection.
+    :type livy_conn_id: str
+    :param polling_interval: time in seconds between polling for job completion. Don't poll for values >=0
+    :type polling_interval: int
+    :param timeout: for a value greater than zero, number of seconds to poll before killing the batch.
+    :type timeout: int
+    """
+
+    @apply_defaults
+    def __init__(
+        self,
+        file=None,
+        args=None,
+        conf=None,
+        livy_conn_id='livy_default',
+        polling_interval=0,
+        timeout=24 * 3600,
+        *vargs,
+        **kwargs
+    ):
+        super(LivyOperator, self).__init__(*vargs, **kwargs)
+
+        self._spark_params = {
+            'file': file,
+            'args': args,
+            'conf': conf,
+        }
+
+        self._spark_params['proxy_user'] = kwargs.get('proxy_user')
+        self._spark_params['class_name'] = kwargs.get('class_name')
+        self._spark_params['jars'] = kwargs.get('jars')
+        self._spark_params['py_files'] = kwargs.get('py_files')
+        self._spark_params['files'] = kwargs.get('files')
+        self._spark_params['driver_memory'] = kwargs.get('driver_memory')
+        self._spark_params['driver_cores'] = kwargs.get('driver_cores')
+        self._spark_params['executor_memory'] = kwargs.get('executor_memory')
+        self._spark_params['executor_cores'] = kwargs.get('executor_cores')
+        self._spark_params['num_executors'] = kwargs.get('num_executors')
+        self._spark_params['archives'] = kwargs.get('archives')
+        self._spark_params['queue'] = kwargs.get('queue')
+        self._spark_params['name'] = kwargs.get('name')
+
+        self._livy_conn_id = livy_conn_id
+        self._polling_interval = polling_interval
+        self._timeout = timeout
+
+        self._livy_hook = None
+        self._batch_id = None
+        self._start_ts = None
+
+    def _init_hook(self):
```

Comment:
This is more commonly called `get_hook` in other operators - can you follow that pattern please?